src/library/librarycontrol.cpp

```diff
@@ -976,6 +990,36 @@ void LibraryControl::slotGoToItem(double v) {
     }
 }
 
+void LibraryControl::slotTriggerItem(double v) {
+    const int triggerItemMethod = static_cast<int>(v);
+
+    // use default gotoitem implementation
+    if (triggerItemMethod == -1) {
+        slotGoToItem(1);
+        return;
+    }
+
+    // check and try toggle items selected on sidebar first if possible
+    if (triggerItemMethod == 0) {
+        if (!m_pSidebarWidget->isLeafNodeSelected()) {
+            setLibraryFocus(FocusWidget::Sidebar);
+            m_pSidebarWidget->toggleSelectedItem();
+        } else
+            slotGoToItem(1);
+        return;
+    }
+
+    // todo: update if focuswidget enum changes
+    // enum overrun
+    if (triggerItemMethod > 5)
+        return;
```

Comment:
```suggestion
    if (triggerItemMethod > static_cast<int>(FocusWidget::Count)) {
        return;
    }
```
(actually it should be limited to the library widgets which are always available (in official skins), but `setLibraryFocus()` already takes care of that